test/test_helper.rb

```diff
@@ -8,6 +8,23 @@
 
 REQUIRED_METADATA_KEYS = %w[topic short_description].freeze
 
+MAX_TOPIC_LENGTH = 35
+
+TOPIC_REGEX = /\A[a-z0-9][a-z0-9-]*\Z/
+
+def invalid_topic_message(topic)
+  "'#{topic}' must be between 1-#{MAX_TOPIC_LENGTH} characters, start with a letter or number, " \
+    "and may include hyphens"
+end
+
+def valid_topic?(raw_topic)
+  return false unless raw_topic
+  normalized_topic = raw_topic.gsub(/\A[[:space:]]+/, "").gsub(/[[:space:]]+\z/, "")
```

Comment:
Maybe personal style but I'd stick each `.gsub` and `.downcase` and `.tr` on a new line?